src/streamlink/plugins/trovo.py

```diff
@@ -0,0 +1,167 @@
+"""
+$description Global video game live streaming and video hosting platform, owned by Tencent.
+$url trovo.live
+$type live, vod
+"""
+
+import logging
+import random
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?trovo\.live/([^/?&]+)(?:/([^/?&]+))?"
+))
+class Trovo(Plugin):
+    @staticmethod
+    def generate_qid():
+        return ("%010x" % random.getrandbits(40)).upper()
+
+    def get_vod(self, video_id):
+        json = self.session.http.post(
+            f"https://gql.trovo.live/?qid={self.generate_qid()}",
+            json=[{
+                "operationName": "batchGetVodDetailInfo",
+                "variables": {
+                    "params": {
+                        "vids": [video_id],
+                    },
+                },
+                "extensions": {
+                    "persistedQuery": {
+                        "version": 1,
+                        "sha256Hash": "ceae0355d66476e21a1dd8e8af9f68de95b4019da2cda8b177c9a2255dad31d0",
+                    },
+                },
+            }],
+            schema=validate.Schema(
+                validate.parse_json(), [{
```

Comment:
Please don't put multiple parameters on the same line like this. Same down below. This will guarenteed cause an unnecessary diff if we switch to a different linting config in the future.